### PR TITLE
Add 1-31 release branch support

### PR DESCRIPTION
**Pull request by sp1999**
*Issue #, if available:*
[#2399](https://github.com/aws/eks-anywhere-internal/issues/2399)

*Description of changes:*
Add EKS-D 1-30 release branch support for build-tooling projects.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: EKSD_LATEST_RELEASES [hunk 1]
```diff
@@ -39,3 +39,7 @@ releases:
 - branch: 1-30
   kubeVersion: v1.30.3
   number: 12
+- branch: 1-31
+  kubeVersion: v1.31.0
+  number: 2
+  dev: true
```
Comment (abhay-krishna) [inline EKSD_LATEST_RELEASES:45]: Why do we need this? EKS Distro 1-31-2 has been released to prod as well.
```console
$ curl https://distro.eks.amazonaws.com/releasechannels/1-31.yaml | yq '.status.latestRelease'
2
```

File: UPSTREAM_PROJECTS.yaml [hunk 2]
```diff
@@ -153,6 +153,8 @@ projects:
             go_version: "1.21"
           - tag: cluster-autoscaler-1.30.1
             go_version: "1.22"
+          - tag: cluster-autoscaler-1.30.1
```
Comment (abhay-krishna) [inline UPSTREAM_PROJECTS.yaml:156]: Can we use the [cluster-autoscaler-1.31.0](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.31.0) tag here?

Comment (sp1999) [reply UPSTREAM_PROJECTS.yaml:156]: I tried using that but the patches didn't apply cleanly. I was able to manually apply the first two patches but the third patch to update go.mod dependencies didn't work and it doesn't look straighforward to update that

Comment (sp1999) [reply UPSTREAM_PROJECTS.yaml:156]: I will upgrade to 1.31 in a separate PR as I certainly need to update the patches especially the third patch where we remove go.mod dependencies. I tried running go mod tidy in the upstream repo checked out with 1.31 tag after applying the first 2 patches but it didn't remove any go dependencies. I will have to check [changelog](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.31.0) for 1.31 and update the patches accordingly.

**Pull request merged**